esrally/driver/runner.py

```diff
@@ -1464,33 +1465,66 @@ async def __call__(self, es, params):
         repository = mandatory(params, "repository", repr(self))
         snapshot = mandatory(params, "snapshot", repr(self))
         body = mandatory(params, "body", repr(self))
-        response = await es.snapshot.create(repository=repository,
-                                            snapshot=snapshot,
-                                            body=body,
-                                            params=request_params,
-                                            wait_for_completion=wait_for_completion)
-
-        # We can derive a more useful throughput metric if the snapshot has successfully completed
-        if wait_for_completion and response.get("snapshot", {}).get("state") == "SUCCESS":
-            stats_response = await es.snapshot.status(repository=repository,
-                                                      snapshot=snapshot)
-            size = stats_response["snapshots"][0]["stats"]["total"]["size_in_bytes"]
-            # while the actual service time as determined by Rally should be pretty accurate, the actual time it took
-            # to restore allows for a correct calculation of achieved throughput.
-            time_in_millis = stats_response["snapshots"][0]["stats"]["time_in_millis"]
-            time_in_seconds = time_in_millis / 1000
-            return {
-                "weight": size,
-                "unit": "byte",
-                "success": True,
-                "service_time": time_in_seconds,
-                "time_period": time_in_seconds
-            }
+        await es.snapshot.create(repository=repository,
+                                 snapshot=snapshot,
+                                 body=body,
+                                 params=request_params,
+                                 wait_for_completion=wait_for_completion)
 
     def __repr__(self, *args, **kwargs):
         return "create-snapshot"
 
 
+class WaitForSnapshotCreate(Runner):
+    async def __call__(self, es, params):
+        repository = mandatory(params, "repository", repr(self))
+        snapshot = mandatory(params, "snapshot", repr(self))
+        wait_period = params.get("completion-recheck-wait-period", 1)
+
+        snapshot_done = False
+        stats = {}
+
+        while not snapshot_done:
+            response = await es.snapshot.status(repository=repository,
+                                                snapshot=snapshot,
+                                                ignore_unavailable=True)
+
+            if "snapshots" in response:
+                response_state = response["snapshots"][0]["state"]
+                # Possible states:
+                # https://www.elastic.co/guide/en/elasticsearch/reference/current/get-snapshot-status-api.html#get-snapshot-status-api-response-body
+                if response_state == "FAILED":
+                    self.logger.error("Snapshot [%s] failed. Response status:\n%s", snapshot, json.dumps(response))
```

Comment:
* "Response status" -> "Response"? (as it is the actual full, response). I also think we should pretty-print it by specifying e.g. `indent=2` in `json.dumps`.

Reply:
Addressed in 32ee4abf6a15d4158992c9843958b51e9a766cb5